src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/DataCenterFormula.kt

```diff
@@ -238,6 +246,57 @@ class DataCenterFormula private constructor(
 
         val provisionedLoadBalancer = futureLoadBalancer.get()
         val loadBalancer = provisionedLoadBalancer.loadBalancer
+
+        val jiraNodeSecurityGroup = jiraStack.findSecurityGroup("JiraNodeSecurityGroup")
```

Comment:
The code that defines all the providers is quite verbose and is partially repeated in StandaloneFormula. Could we do something about it?

Reply:
We could, but I'm not sure if it's worth it right now.

I think that both of those formulas should be refactored and split into smaller responsibilities eventually, however it may be easier to that in bulk for all of the instructions there. I had some thoughts around it when i was experimenting around https://stash.atlassian.com/projects/HL/repos/jpt-api-reimpl-spike

Reply:
- [ ] Refactor new content of Formulas to be more readable

Reply:
I can see "JiraNodeSecurityGroup" is used 2 times in this class, maybe extract it to const?

Reply:
Out of all suggestions this one I didn't manage to implement.

My idea is to break down the responsibilities of `DataCenterFormula` into it's smaller dependencies (interfaces) which could be overwritten, but would have their default implementations (classes). We could split things like:
 * Provisioner
 * Instance Identifier
 * Ssh/IP Resolver
 * Access Definition (this one would be the one around my changes)

I don't have time to finish it and the refactoring should be possible later on, so I'd like to proceed without it.